crates/services/gas_price_service/src/v1/da_source_service/block_committer_costs.rs

```diff
@@ -142,54 +142,315 @@ impl BlockCommitterHttpApi {
     }
 }
 
+const PAGE_SIZE: u32 = 10;
 #[async_trait::async_trait]
 impl BlockCommitterApi for BlockCommitterHttpApi {
-    async fn get_latest_costs(&self) -> DaBlockCostsResult<Option<RawDaBlockCosts>> {
+    async fn get_costs_by_l2_block_number(
+        &self,
+        l2_block_number: u32,
+    ) -> DaBlockCostsResult<Vec<RawDaBlockCosts>> {
+        // Specific: http://localhost:8080/v1/costs?variant=specific&value=19098935&limit=5
         if let Some(url) = &self.url {
-            let val = self.client.get(url).send().await?;
+            let formatted_url = format!("{url}/v1/costs?variant=specific&value={l2_block_number}&limit={PAGE_SIZE}");
+            let val = self.client.get(formatted_url).send().await?;
             tracing::warn!("val: {:?}", val);
-            let response = val.json::<Option<RawDaBlockCosts>>().await?;
+            let response = val.json::<Vec<RawDaBlockCosts>>().await?;
             tracing::warn!("Response: {:?}", response);
             Ok(response)
         } else {
-            Ok(None)
+            Ok(vec![])
         }
     }
 
-    async fn get_costs_by_seqno(
-        &self,
-        number: u32,
-    ) -> DaBlockCostsResult<Option<RawDaBlockCosts>> {
+    async fn get_latest_costs(&self) -> DaBlockCostsResult<Option<RawDaBlockCosts>> {
+        // Latest: http://localhost:8080/v1/costs?variant=latest&limit=5
         if let Some(url) = &self.url {
-            let val = self
-                .client
-                .get(format!("{}/{}", url, number))
-                .send()
-                .await?;
-            tracing::warn!("val: {:?}", val);
-            let response = val.json::<Option<RawDaBlockCosts>>().await?;
-            tracing::warn!("Response: {:?}", response);
-            Ok(response)
+            let formatted_url = format!("{url}/v1/costs?variant=latest&limit=1");
+            let response = self.client.get(formatted_url).send().await?;
+            tracing::warn!("val: {:?}", response);
+            let raw_da_block_costs = response.json::<Vec<RawDaBlockCosts>>().await?;
+            tracing::warn!("Response: {:?}", raw_da_block_costs);
+            // only take the first element, since we are only looking for the most recent
+            Ok(raw_da_block_costs.first().cloned())
```

Comment:
Just sanity checking here, we're getting the responses in descending (relative to block height or bundle ID) order?

Reply:
in the latest endpoint, yes